src/igvfd/tests/test_upgrade_analysis_set.py

```diff
@@ -13,3 +13,9 @@ def test_analysis_set_upgrade_1_2(upgrader, analysis_set_v1):
     assert 'input_file_set' not in value
     assert input_file_sets == value['input_file_sets']
     assert value['schema_version'] == '2'
+
+
+def test_analysis_set_upgrade_3_4(upgrader, analysis_set_v3):
+    value = upgrader.upgrade('analysis_set', analysis_set_v3, current_version='3', target_version='4')
```

Comment:
please add validation of preexisting file_set_type

Reply:
added